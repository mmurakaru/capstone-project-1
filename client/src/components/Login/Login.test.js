import { render } from '@testing-library/react'
import Login from './Login'

describe('Login', () => {
  it('renders correctly', () => {
    const { container } = render(<Login />)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('links to the right location by clicking the login-button', () => {
    const { getByTestId } = render(<Login />)
    expect(getByTestId('login-button').href).toBe('http://localhost:3001/login')
  })
})
