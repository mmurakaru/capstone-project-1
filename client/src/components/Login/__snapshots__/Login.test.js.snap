// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
.c0 {
  background: var(--gradient-regular);
  height: 100vh;
}

.c1 {
  display: grid;
  grid-template-rows: 2fr 1fr;
  height: 100vh;
  padding: 0 44px;
}

.c2 {
  grid-row-start: 1;
  place-self: center;
  width: 90%;
}

.c3 {
  grid-row-start: 2;
}

.c3 p {
  color: white;
  font-size: 0.9;
  font-weight: 300;
  text-align: center;
}

.c3 span {
  font-weight: 700;
}

.c3 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4 {
  background: var(--white);
  border-radius: 28px;
  box-shadow: var(--shadow-dark);
  padding: 6px 0 12px;
  width: 100%;
}

.c4 p {
  color: var(--spotify-green);
  font-weight: 700;
  margin: auto;
}

.c5 {
  width: 22px;
  position: relative;
  top: 4.5px;
  margin-left: 2px;
}

<section
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <svg>
        resonance.svg
      </svg>
    </div>
    <div
      class="c3"
    >
      <a
        data-testid="login-button"
        href="http://localhost:3001/login"
      >
        <div
          class="c4"
        >
          <p>
            Login with 
            <svg
              class="c5"
            >
              spotify.svg
            </svg>
          </p>
        </div>
      </a>
      <p>
        and discover new songs 
        <br />
         and artists.
      </p>
    </div>
  </div>
</section>
`;
